pkg/resources/schema.go

```diff
@@ -29,268 +33,380 @@ var schemaSchema = map[string]*schema.Schema{
-		log.Printf("[DEBUG] schema (%s) not found", d.Id())
-		d.SetId("")
-		return nil
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to query secondary database. Marking the resource as removed.",
+					Detail:   fmt.Sprintf("DatabaseName: %s, Err: %s", id.FullyQualifiedName(), err),
+				},
+			}
+		}
+		return diag.FromErr(err)
+	}
+	if err := d.Set("name", schema.Name); err != nil {
+		return diag.FromErr(err)
 	}
 
-	var retentionTime int64
-	// "retention_time" may sometimes be empty string instead of an integer
-	{
-		rt := s.RetentionTime
-		if rt == "" {
-			rt = "0"
-		}
+	if err := d.Set("is_transient", schema.IsTransient()); err != nil {
+		return diag.FromErr(err)
+	}
 
-		retentionTime, err = strconv.ParseInt(rt, 10, 64)
-		if err != nil {
-			return err
-		}
+	if err := d.Set("with_managed_access", schema.IsManagedAccess()); err != nil {
+		return diag.FromErr(err)
 	}
 
-	if dataRetentionDays := d.Get("data_retention_days"); dataRetentionDays.(int) != IntDefault || int64(database.RetentionTime) != retentionTime {
-		if err := d.Set("data_retention_days", retentionTime); err != nil {
-			return err
-		}
+	if err := d.Set("database", schema.DatabaseName); err != nil {
+		return diag.FromErr(err)
 	}
 
-	values := map[string]any{
-		"name":     s.Name,
-		"database": s.DatabaseName,
-		"comment":  s.Comment,
-		// reset the options before reading back from the DB
-		"is_transient": false,
-		"is_managed":   false,
+	if err := d.Set("comment", schema.Comment); err != nil {
+		return diag.FromErr(err)
 	}
 
-	for k, v := range values {
-		if err := d.Set(k, v); err != nil {
-			return err
-		}
+	schemaParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+		In: &sdk.ParametersIn{
+			Schema: id,
+		},
+	})
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
-	if opts := s.Options; opts != nil && *opts != "" {
-		for _, opt := range strings.Split(*opts, ", ") {
-			switch opt {
-			case "TRANSIENT":
-				if err := d.Set("is_transient", true); err != nil {
-					return err
-				}
-			case "MANAGED ACCESS":
-				if err := d.Set("is_managed", true); err != nil {
-					return err
-				}
-			}
-		}
+	if diags := HandleDatabaseParameterRead(d, schemaParameters); diags != nil {
+		return diags
+	}
+	pipeExecutionPaused, err := collections.FindOne(schemaParameters, func(property *sdk.Parameter) bool {
+		return property.Key == "PIPE_EXECUTION_PAUSED"
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to find schema PIPE_EXECUTION_PAUSED parameter, err = %w", err))
+	}
+	value, err := strconv.ParseBool((*pipeExecutionPaused).Value)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("pipe_execution_paused", value); err != nil {
+		return diag.FromErr(err)
 	}
 
+	if err = d.Set(ShowOutputAttributeName, []map[string]any{schemas.SchemaToSchema(schema)}); err != nil {
```

Comment:
Are we intentionally not adding the Describe output to the schema?

Reply:
Added.